<?xml version="1.0" ?>
<robot name="m10ia_on_gantry" xmlns:xacro="http://ros.org/wiki/xacro">
  <xacro:include filename="$(find fanuc_m10ia_support)/urdf/m10ia_macro.xacro" />

  <group name="robot">
    <chain base_link="robot_base_link" tip_link="robot_tool0" />
  </group>

  <group name="gantry">
    <joint name="carriage_joint_1" />
    <joint name="carriage_joint_2" />
    <joint name="carriage_joint_3" />
  </group>

  <group name="robot_with_gantry">
    <group name="robot" />
    <group name="gantry" />
  </group>

  <group name="welding_torch">
    <link name="torch_link" />
    <link name="nozzle_link" />
  </group>

  <end_effector name="welding_torch" parent_link="robot_tool0" group="welding_torch" parent_group="robot" />

  <disable_collisions link1="base_link" link2="carriage_link_1" />
  <disable_collisions link1="carriage_link_1" link2="carriage_link_2" />
  <disable_collisions link1="carriage_link_1" link2="carriage_link_3" />
  <disable_collisions link1="carriage_link_2" link2="carriage_link_3" />

  <disable_collisions link1="carriage_link_3" link2="robot_base_link" />

  <disable_collisions link1="robot_base_link" link2="robot_link_1" />
  <disable_collisions link1="robot_base_link" link2="robot_link_2" />
  <disable_collisions link1="robot_link_1" link2="robot_link_2" />
  <disable_collisions link1="robot_link_1" link2="robot_link_3" />
  <disable_collisions link1="robot_link_2" link2="robot_link_3" />
  <disable_collisions link1="robot_link_3" link2="robot_link_4" />
  <disable_collisions link1="robot_link_3" link2="robot_link_5" />
  <disable_collisions link1="robot_link_3" link2="robot_link_6" />
  <disable_collisions link1="robot_link_4" link2="robot_link_5" />
  <disable_collisions link1="robot_link_5" link2="robot_link_6" />

  <disable_collisions link1="robot_link_5" link2="torch_link" />
  <disable_collisions link1="robot_link_5" link2="nozzle_link" />
  <disable_collisions link1="robot_link_6" link2="torch_link" />
  <disable_collisions link1="robot_link_6" link2="nozzle_link" />
  <disable_collisions link1="torch_link" link2="nozzle_link" />
</robot>
